src/Engine/ProtoCore/DSASM/Executable.cs

```diff
@@ -141,7 +141,7 @@ public class CodeBlock
         /// <param name="procTable"></param>
         /// <param name="isBreakableBlock"></param>
         /// <param name="core"></param>
-        public CodeBlock(Guid guid, CodeBlockType type, ProtoCore.Language langId, int cbID, SymbolTable symbols, ProcedureTable procTable, bool isBreakableBlock = false, ProtoCore.Core core = null)
+        public CodeBlock(Guid guid, CodeBlockType type, ProtoCore.Language langId, int cbID, SymbolTable symbols, ProcedureTable procTable, bool isBreakableBlock, ProtoCore.Core core)
```

Comment:
this is a breaking change.

Reply:
ok, you are correct, I will revert it. However, these default arguments are used in the code without being checked for being null. It looks like the code has been this way for a while, not sure what you would like to happen here (other than no changes).